universal-application-tool-0.0.1/app/repository/VersionRepository.java

```diff
@@ -285,11 +283,17 @@ protected PredicateExpressionNode updatePredicateNode(PredicateExpressionNode cu
     }
   }
 
-  public void updateProgramsForNewDraftQuestion(long oldId) {
+  /**
+   * Update all ACTIVE and DRAFT programs that refer to the question revision {@code oldId}, to
+   * refer to the latest revision of all their questions.
+   */
+  public void updateProgramsThatReferenceQuestion(long oldId) {
+    // Update all DRAFT programs that reference the question.
     getDraftVersion().getPrograms().stream()
         .filter(program -> program.getProgramDefinition().hasQuestion(oldId))
-        .forEach(program -> updateQuestionVersions(program));
+        .forEach(this::updateQuestionVersions);
 
+    // Update any ACTIVE program that references the question and does not have a DRAFT already
```

Comment:
question (blocking): updates the active program revision or creates a new draft and updates it? I think it's the latter

Reply:
Updated PTAL. There's a nomenclature trick in that "updating a ACTIVE" has the system-design property that it result in a draft.

Reply:
LGTM!